site/src/pages/TerminalPage/TerminalPage.tsx

```diff
@@ -309,12 +289,62 @@ const TerminalPage: FC<
           </Stack>
         )}
       </div>
+      {shouldDisplayStartupWarning && (
+        <div className={styles.alert} role="alert">
+          <WarningIcon className={styles.alertIcon} />
+          <div>
+            <div className={styles.alertTitle}>
+              Startup script is still running
+            </div>
+            <div className={styles.alertMessage}>
+              You can use it but dotfiles aren&lsquo;t setup yet
+            </div>
+          </div>
+          <div className={styles.alertActions}>
+            <Button
+              startIcon={<RefreshOutlined />}
+              size="small"
+              onClick={() => {
+                window.location.reload()
+              }}
+            >
+              Refresh
+            </Button>
+          </div>
+        </div>
+      )}
       <div className={styles.terminal} ref={xtermRef} data-testid="terminal" />
     </>
   )
 }
 
-export default TerminalPage
+const useReloading = (isDisconnected: boolean) => {
+  const [status, setStatus] = useState<"reloading" | "notReloading">(
+    "notReloading",
+  )
+
+  // Retry connection on key press when it is disconnected
+  useEffect(() => {
+    if (!isDisconnected) {
+      return
+    }
+
+    const keyDownHandler = () => {
+      setStatus("reloading")
+      window.location.reload()
+    }
+
+    document.addEventListener("keydown", keyDownHandler)
```

Comment:
Question: Should we use 'keypress' instead of keydown?  My understanding is with keydown, the event handler will be called numerous times while the key is being pressed. Is that the desired behavior?

Reply:
I think so, but I'm not 100% sure. Since it is an "old" behavior and we didn't have complainings about that I would let it as it is for now.